katharsis-core/src/main/java/io/katharsis/queryspec/InMemoryEvaluator.java

```diff
@@ -97,11 +97,15 @@ private static boolean matchesPrimitiveOperator(Object object, FilterSpec filter
 		Object value = PropertyUtils.getProperty(object, filterSpec.getAttributePath());
 		FilterOperator operator = filterSpec.getOperator();
 		Object filterValue = filterSpec.getValue();
-		if (value instanceof Collection) {
-			return matchesAny((Collection<?>) value, operator, filterValue);
-		}
-		else {
-			return operator.matches(value, filterValue);
+		if (filterValue != null ||  (filterValue == null && (operator.equals(FilterOperator.EQ) || operator.equals(FilterOperator.NEQ)))) {
```

Comment:
Katharsis library requires minimum Java 7 to build and run therefore lambda expressions cannot be used.